fire/cli/mtl.py

```diff
@@ -841,3 +908,356 @@ def regn(projektnavn: str, **kwargs) -> None:
+    nye_punkter = set(nyetablerede.index)
+    nyetablerede = nyetablerede.reset_index()
+    n = nyetablerede.shape[0]
+
+    if n == 0:
+        fire.cli.print("Ingen nyetablerede punkter at registrere")
+        return
+
+    v = Koordinatvrider()
+    til_registrering = []
+    anvendte_landsnumre = {}
+    sagsevent = None
+    landsnummer_pit = firedb.hent_punktinformationtype("IDENT:landsnr")
+    beskrivelse_pit = firedb.hent_punktinformationtype("ATTR:beskrivelse")
+    h_over_terræn_pit = firedb.hent_punktinformationtype("AFM:højde_over_terræn")
+    assert landsnummer_pit is not None, "Rådden pit"
+    assert beskrivelse_pit is not None, "Rådden pit"
+    assert h_over_terræn_pit is not None, "Rådden pit"
+
+    # Vi samler de genererede punkter i en dict, så de kan persisteres samlet
+    # under et enkelt sagsevent
+    genererede_punkter = {}
+    genererede_landsnumre = []
+
+    for i in range(n):
+        # Et tomt tekstfelt kan repræsenteres på en del forskellige måder...
+        # Punkter udstyret med uuid er allerede registrerede
+        # if not (nyetablerede["uuid"][i] in ["", None] or pd.isna(nyetablerede["uuid"][i])):
+        if str(nyetablerede.uuid[i]) not in ["", "None", "nan"]:
+            continue
+        print(f"Behandler punkt {nyetablerede['Foreløbigt navn'][i]}")
+
+        lokation = v.vrid(nyetablerede["λ"][i], nyetablerede["φ"][i])
+        distrikt = nyetablerede["Landsnummer"][i]
+
+        # Gør klar til at finde et ledigt landsnummer, hvis vi ikke allerede har et
+        if 2 == len(distrikt.split("-")):
+            if distrikt in anvendte_landsnumre:
+                numre = anvendte_landsnumre[distrikt]
+            else:
+                numre = set(find_alle_landsnumre_i_distrikt(distrikt))
+                anvendte_landsnumre[distrikt] = numre
+                print(f"Fandt {len(numre)} punkter i distrikt {distrikt}")
+        # Hvis der er anført et fuldt landsnummer må det hellere se ud som et
+        elif 3 != len(distrikt.split("-")):
+            fire.cli.print(f"Usselt landsnummer: {distrikt}")
+            continue
+        # Ellers har vi et komplet landsnummer, så punktet er allerede registreret
+        else:
+            continue
+
+        # Hjælpepunkter har egen nummerserie
+        if "ingen" == str(nyetablerede["Afmærkning"][i]).lower():
+            nummerserie = range(90001, 99999)
+        else:
+            nummerserie = itertools.chain(range(9001, 10000), range(19001, 20000))
+
+        # Så leder vi...
+        for løbenummer in nummerserie:
+            if løbenummer not in numre:
+                landsnummer = f"{distrikt}-{løbenummer:05}"
+                genererede_landsnumre.append(landsnummer)
+                fire.cli.print(f"Anvender landsnummer {landsnummer}")
+                numre.add(løbenummer)
+                break
+        else:
+            # for/else er sjældent brugt, men nyttig her: eksekveres når
+            # for-løkken IKKE afbrydes af break
```

Comment:
Man kan godt argumentere for at det bør en pythonprogrammør vide, og at kommentaren derfor er overflødig

Reply:
Konstruktionen er jo kontroversiel og jeg har aldrig set den brugt andre steder, men jeg har rettet til "Hvis for-løkken løber til ende er vi løbet tør for løbenumre", som i højere grad beskriver problemet, frem for udførelsen